### PR TITLE
Cerberus pod/related configmaps specs to monitor Prow Controller Health

### PR DESCRIPTION
**Pull request by Rajalakshmi-Girish**
This PR adds the spec files required for deploying Cerberus pod and the related config maps. 
Below are the config Map details:
`cerberus-config` : Kubernetes related configuration required by cerberus
`cerberus-custom-check` : File for checking Failed message in the prow controller log
`cerberus-start` : Contains the changes to start_cerberus.py to enable custom check notifications to slack
`cerberus-slack-client` : Changes to display custom check failure messages to slack.

Configmaps `cerberus-start` and `cerberus-slack-client` can be removed once the below PR merges:
https://github.com/cloud-bulldozer/cerberus/pull/119

_**Note:**_ **The pod spec is not having any configuration exported for the underlying cluster
It is assumed that pod run in-cluster using the `default` service account in the `prow` namespace will be able to access logs of other pods in `prow` namespace.**

### CODE REVIEW

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ppc64le-cloud/test-infra/pull/140#" title="Author self-approved">Rajalakshmi-Girish</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **mkumatag** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @mkumatag` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ppc64le-cloud%2Ftest-infra).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/ppc64le-cloud/test-infra/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by ltccci** [comment]
@Rajalakshmi-Girish: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pull-test-infra-yamllint | b69f4b2e8c8a267b2e9ae7927d9418eab78ffbdf | [link](https://prow.ppc64le-cloud.org/view/s3/prow-logs/pr-logs/pull/ppc64le-cloud_test-infra/140/pull-test-infra-yamllint/1362398654761013248) | `/test pull-test-infra-yamllint`

[Full PR test history](https://prow.ppc64le-cloud.org/pr-history?org=ppc64le-cloud&repo=test-infra&pr=140). [Your PR dashboard](https://prow.ppc64le-cloud.org/pr?query=is:pr+state:open+author:Rajalakshmi-Girish).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by ltccci** [comment]
@Rajalakshmi-Girish: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pull-test-infra-yamllint | 64e716c13320bf1b12e91f68214879a5b78b6f1a | [link](https://prow.ppc64le-cloud.org/view/s3/prow-logs/pr-logs/pull/ppc64le-cloud_test-infra/140/pull-test-infra-yamllint/1362399209474494464) | `/test pull-test-infra-yamllint`

[Full PR test history](https://prow.ppc64le-cloud.org/pr-history?org=ppc64le-cloud&repo=test-infra&pr=140). [Your PR dashboard](https://prow.ppc64le-cloud.org/pr?query=is:pr+state:open+author:Rajalakshmi-Girish).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by ltccci** [comment]
@Rajalakshmi-Girish: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pull-test-infra-yamllint | c5b90ccdc3f519f25c01c5990949b152cf3db423 | [link](https://prow.ppc64le-cloud.org/view/s3/prow-logs/pr-logs/pull/ppc64le-cloud_test-infra/140/pull-test-infra-yamllint/1362399703815163904) | `/test pull-test-infra-yamllint`

[Full PR test history](https://prow.ppc64le-cloud.org/pr-history?org=ppc64le-cloud&repo=test-infra&pr=140). [Your PR dashboard](https://prow.ppc64le-cloud.org/pr?query=is:pr+state:open+author:Rajalakshmi-Girish).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by ltccci** [comment]
@Rajalakshmi-Girish: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pull-test-infra-yamllint | c5b90ccdc3f519f25c01c5990949b152cf3db423 | [link](https://prow.ppc64le-cloud.org/view/s3/prow-logs/pr-logs/pull/ppc64le-cloud_test-infra/140/pull-test-infra-yamllint/1362460288040833024) | `/test pull-test-infra-yamllint`

[Full PR test history](https://prow.ppc64le-cloud.org/pr-history?org=ppc64le-cloud&repo=test-infra&pr=140). [Your PR dashboard](https://prow.ppc64le-cloud.org/pr?query=is:pr+state:open+author:Rajalakshmi-Girish).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: config/prow/cluster/cerberus.yaml
Comment (mkumatag) [reply config/prow/cluster/cerberus.yaml:8]: Make sure you test it in one of your local environment whether it is really working or not.

**Review by mkumatag — CHANGES_REQUESTED** [review]

**Comment by ltccci** [comment]
@Rajalakshmi-Girish: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pull-test-infra-yamllint | 6f44fece8ab5f1b26cf42dbdf85a598e07ac18cc | [link](https://prow.ppc64le-cloud.org/view/s3/prow-logs/pr-logs/pull/ppc64le-cloud_test-infra/140/pull-test-infra-yamllint/1366833821692989440) | `/test pull-test-infra-yamllint`

[Full PR test history](https://prow.ppc64le-cloud.org/pr-history?org=ppc64le-cloud&repo=test-infra&pr=140). [Your PR dashboard](https://prow.ppc64le-cloud.org/pr?query=is:pr+state:open+author:Rajalakshmi-Girish).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: config/prow/cluster/cerberus_configmaps.yaml
Comment (Rajalakshmi-Girish) [inline config/prow/cluster/cerberus_configmaps.yaml:12]: Disabling `watch_nodes` as we only intended to see `prow-controller` logs.

File: config/prow/cluster/cerberus_configmaps.yaml
Comment (Rajalakshmi-Girish) [inline config/prow/cluster/cerberus_configmaps.yaml:63]: Picking up the `prow controller` pod name and getting `logs` for it to see if the error string is present.

File: config/prow/cluster/cerberus_configmaps.yaml
Comment (Rajalakshmi-Girish) [inline config/prow/cluster/cerberus_configmaps.yaml:279]: Commenting call to `check_namespaces` as we do not intend to give _list all namespaces_ role to cerberus sa.

File: config/prow/cluster/cerberus_configmaps.yaml
Comment (Rajalakshmi-Girish) [inline config/prow/cluster/cerberus_configmaps.yaml:287]: Commenting this command to invoke `cluster-info` as this needs _role to get services_ resource!

File: config/prow/cluster/cerberus_configmaps.yaml
Comment (Rajalakshmi-Girish) [inline config/prow/cluster/cerberus_configmaps.yaml:319]: Commenting call to `list nodes `as this needs sa to have a _role to list nodes._

**Comment by ltccci** [comment]
@Rajalakshmi-Girish: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pull-test-infra-yamllint | 7414c584f3cf4af72ff62f4e8c091e8662fb5680 | [link](https://prow.ppc64le-cloud.org/view/s3/prow-logs/pr-logs/pull/ppc64le-cloud_test-infra/140/pull-test-infra-yamllint/1367335069659697152) | `/test pull-test-infra-yamllint`

[Full PR test history](https://prow.ppc64le-cloud.org/pr-history?org=ppc64le-cloud&repo=test-infra&pr=140). [Your PR dashboard](https://prow.ppc64le-cloud.org/pr?query=is:pr+state:open+author:Rajalakshmi-Girish).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by Rajalakshmi-Girish** [comment]
I have added `ClusterRoles` to `cereberus_rbac.yaml` to list/get nodes, namespaces, services. Thus have modified the `start_cerberus.py` accordingly to match with the upstream file.
The configmaps `cerberus-slack-client` and `cerberus-start ` can be removed once the upstream PR merges.
https://github.com/cloud-bulldozer/cerberus/pull/119

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ppc64le-cloud/test-infra/pull/140#issuecomment-792457048" title="Approved">mkumatag</a>*, *<a href="https://github.com/ppc64le-cloud/test-infra/pull/140#" title="Author self-approved">Rajalakshmi-Girish</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ppc64le-cloud%2Ftest-infra).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ppc64le-cloud/test-infra/blob/master/OWNERS)~~ [mkumatag]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by mkumatag** [comment]
Lemme deploy this service and see how it goes,

**Pull request merged**
